modules/library/jdbc/src/main/java/org/geotools/jdbc/MetadataTablePrimaryKeyFinder.java

```diff
@@ -112,17 +113,26 @@ public PrimaryKey getPrimaryKey(JDBCDataStore store, String schema, String table
                 synchronized (this) {
                     if (metadataTableExists == null) {
                         try {
+                            // This is a database-agnostic way to check if the table exists without
+                            // resulting in error messages. It also checks for case sensitivity.
+                            tablesRs =
+                                    cx.getMetaData()
+                                            .getTables(
+                                                    null,
+                                                    metadataSchema,
+                                                    "%",
+                                                    new String[] {"TABLE", "VIEW"});
 
-                            StringBuffer sb = new StringBuffer();
-                            sb.append("SELECT * FROM ");
-                            if (metadataSchema != null) {
-                                store.getSQLDialect().encodeSchemaName(metadataSchema, sb);
-                                sb.append(".");
+                            metadataTableExists = false;
+
+                            while (tablesRs.next()) {
+                                String tblName = tablesRs.getString(3);
+                                if (tblName.toLowerCase().equals(tableName.toLowerCase())) {
+                                    metadataTableExists = true;
+                                    break;
+                                }
                             }
-                            sb.append(tableName).append(" WHERE 1 = 0");
-                            st = cx.prepareStatement(sb.toString());
-                            rs = st.executeQuery();
-                            metadataTableExists = true;
+
                         } catch (Exception e) {
                             // clean up the transaction status in case we are in auto-commit mode
                             if (e instanceof SQLException && !cx.getAutoCommit()) {
```

Comment:
I would add a WARNING level log with the exception.

Reply:
Done.